packages/backend/server/src/core/auth/session.ts

```diff
@@ -0,0 +1,300 @@
+import { randomUUID } from 'node:crypto';
+
+import {
+  Injectable,
+  NotAcceptableException,
+  NotFoundException,
+} from '@nestjs/common';
+import { verify } from '@node-rs/argon2';
+import { PrismaClient, Session } from '@prisma/client';
+import { CookieOptions } from 'express';
+import { omit } from 'lodash-es';
+
+import { Config, SessionCache } from '../../fundamentals';
+
+export function parseAuthUserSeqNum(value: any) {
+  switch (typeof value) {
+    case 'number': {
+      return value;
+    }
+    case 'string': {
+      value = Number.parseInt(value);
+      return Number.isNaN(value) ? 0 : value;
+    }
+
+    default: {
+      return 0;
+    }
+  }
+}
+
+@Injectable()
+export class SessionService {
+  readonly sessionCookieName = 'sid';
+  readonly authUserSeqCookieName = 'authuser';
+
+  readonly cookieOptions: CookieOptions = {
+    sameSite: 'lax',
+    httpOnly: true,
+    path: '/',
+    domain: this.config.host,
+    secure: this.config.https,
+  };
+
+  constructor(
+    private readonly db: PrismaClient,
+    private readonly cache: SessionCache,
+    private readonly config: Config
+  ) {}
+
+  async signIn(email: string, password: string, existingSession?: string) {
+    const user = await this.db.user.findFirst({
+      where: {
+        email: {
+          equals: email,
+          mode: 'insensitive',
+        },
+      },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User Not Found');
+    }
+
+    if (!user.password) {
+      throw new NotAcceptableException(
+        'User Password is not set. Should login throw email link.'
+      );
+    }
+
+    const passwordMatches = await verify(user.password, password);
```

Comment:
add a length check for password to prevent dos attack

Reply:
should be checked in user input

Reply:
I mean the input in resolver or controller, not in service

Reply:
make sence